### PR TITLE
feat(historical): skip null rows

### LINKED ISSUE #208: Handle 'null' numbers when importing
# Bug Report

## Describe the bug
Symbol [EURGBP=X](https://uk.finance.yahoo.com/quote/EURGBP%3DX/history?period1=1570492800&period2=1570579200&interval=1d&filter=history&frequency=1d&includeAdjustedClose=true) has no records for '2019-10-09', so when queried historically is filled with nulls (check the .csv import from the link).

I'd expect the numbers to be `null` not the stringified `'null'`.

## Minimal Reproduction
```typescript
const queryOptions = { period1: '2019-10-08', period2: '2019-10-09' };
const res = await yahooFinance.historical(symbol, queryOptions, { validateResult: false });
console.log(res);
```
produces:
```typescript
[
  {
    date: 2019-10-08T00:00:00.000Z,
    open: 0.89283,
    high: 0.89988,
    low: 0.8922,
    close: 0.8928,
    adjClose: 0.8928,
    volume: 0
  },
  {
    date: 2019-10-09T00:00:00.000Z,
    open: 'null',
    high: 'null',
    low: 'null',
    close: 'null',
    adjClose: 'null',
    volume: 'null'
  }
]
```
The same output with 'null' is produced when the result is validated.

## Environment
Node version (if applicable): 14.17.0
yarn version: 1.22.10
Library version (e.g. 1.10.1): 1.12.1

### PR DESCRIPTION
**Pull request by gadicc**
Closes #208.

## Changes
- csv2json: convert "null" -> null;
- historical: skip rows with all nulls, throw on some nulls.

### CODE REVIEW

File: src/modules/historical.ts [hunk 1]
```diff
@@ -80,6 +80,44 @@ export default function historical(
...
     result: {
       schemaKey: "#/definitions/HistoricalResult",
+      transformWith(result: any) {
+        const filteredResults = [];
+        const fieldCount = Object.keys(result[0]).length;
+
+        // Count number of null values in object (1-level deep)
+        function nullFieldCount(object: Object) {
+          let nullCount = 0;
+          for (let val of Object.values(object)) if (val === null) nullCount++;
```
Comment (okomarov) [inline src/modules/historical.ts:90]: nit: const val

File: src/modules/historical.ts [hunk 2]
```diff
@@ -80,6 +80,44 @@ export default function historical(
...
+          return nullCount;
+        }
+
+        for (let row of result) {
```
Comment (okomarov) [inline src/modules/historical.ts:94]: nit: const row

**Comment by okomarov** [comment]
Functionality-wise looks good :)

Not familiar with the:
```
schemaKey: "#/definitions/HistoricalResult",
      transformWith(result: any)
```
but probably a way to ensure `result: any` conforms with some interface?

File: src/modules/historical.ts (hunk 1 above)
Comment (gadicc) [reply src/modules/historical.ts]: thanks, @okomarov, this is an old bad habit of mine that I seem to always slip back into without realising.  so thanks for that.

File: src/modules/historical.ts (hunk 2 above)
Comment (gadicc) [reply src/modules/historical.ts]: ditto

**Comment by gadicc** [comment]
Thanks, @okomarov - appreciated.

Yup, that's correct... `transformWith()` takes the original response from Yahoo and modifies it slightly before passing it on to the runtime validation and type coersion (using the specified schema).  The typescript is set up so that ultimately the correct types are returned.

In case you're interested, take a look at [moduleExec.ts](https://github.com/gadicc/node-yahoo-finance2/blob/devel/src/lib/moduleExec.ts) - which is the common code used by all the modules.  There's a summary comment at the very top.

**Pull request merged**